tools/autograd/templates/Functions.cpp

```diff
@@ -726,20 +726,10 @@ Tensor _fused_dropout_backward(Tensor grad, Tensor mask, double p1m) {
   }
 }
 
-Tensor select_first_equal_backward(Tensor grad, const Tensor & input, const Tensor & value) {
-  auto grad_input = at::zeros_like(input);
-
-  // find indices of the first element for which input[idx] == value
-  auto first_value_idx = (input == value).nonzero().select(0, 0);
-
-  if (grad_input.dim() == 0) {
-    grad_input.copy_(grad);
-  }
-  else {
-    grad_input.index_put_(at::chunk(first_value_idx, grad_input.dim()), grad);
-  }
-
-  return grad_input;
+Tensor evenly_dispatch_backward(Tensor grad, const Tensor & input, const Tensor & value) {
+  auto mask = (input == value);
+  auto count = mask.sum(input.scalar_type());
```

Comment:
For tensors with >2^24 elements that all happen to be max/min/median, this is going to be inaccurate, so maybe better leave in int64 or double, depending on which is faster?

Reply:
It is a scalar, so I don't think it makes any difference on int64 vs fp64. Let's just use the default (int64).